ksql-rest-app/src/test/java/io/confluent/ksql/rest/integration/AuthorizationFunctionalTest.java

```diff
@@ -0,0 +1,210 @@
+
+import javax.ws.rs.core.Configurable;
+import java.security.Principal;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static io.confluent.ksql.test.util.EmbeddedSingleNodeKafkaCluster.JAAS_KAFKA_PROPS_NAME;
+import static io.confluent.ksql.test.util.EmbeddedSingleNodeKafkaCluster.VALID_USER1;
+import static io.confluent.ksql.test.util.EmbeddedSingleNodeKafkaCluster.VALID_USER2;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@Category({IntegrationTest.class})
+public class AuthorizationFunctionalTest {
+  private static final IntegrationTestHarness TEST_HARNESS = IntegrationTestHarness.build();
+
+  private static final TestKsqlRestApp REST_APP = TestKsqlRestApp
+      .builder(TEST_HARNESS::kafkaBootstrapServers)
+      .withProperty("authentication.method", "BASIC")
+      .withProperty("authentication.roles", "**")
+      // Reuse the Kafka JAAS config for KSQL authentication which has the same valid users
+      .withProperty("authentication.realm", JAAS_KAFKA_PROPS_NAME)
+      .withProperty(
+          KsqlConfig.KSQL_SECURITY_EXTENSION_CLASS,
+          MockKsqlSecurityExtension.class.getName()
+      )
+      .withServiceContextBinder(config -> new AbstractBinder() {
+        @Override
+        protected void configure() {
+          bindFactory(new Factory<ServiceContext>() {
+            @Override
+            public ServiceContext provide() {
+              return TEST_HARNESS.getServiceContext();
+            }
+
+            @Override
+            public void dispose(final ServiceContext serviceContext) {
+              // do nothing because TEST_HARNESS#getServiceContext always
+              // returns the same instance
+            }
+          })
+              .to(ServiceContext.class)
+              .in(RequestScoped.class);
+        }
+      })
+      .build();
+
+  private static Set<String> allowedUsers = new HashSet<>();
+
+  @Mock
+  private KsqlAuthorizationProvider authorizationProvider;
+
+  @ClassRule
+  public static final RuleChain CHAIN = RuleChain.outerRule(TEST_HARNESS).around(REST_APP);
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    allowedUsers.clear();
+    MockKsqlSecurityExtension.setAuthorizationProvider(authorizationProvider);
+  }
+
+  @Test
+  public void shouldLoadSecurityExtension() {
+    // Then:
+    assertThat(MockKsqlSecurityExtension.initialized, is(true));
+    assertThat(MockKsqlSecurityExtension.closed, is(false));
+  }
+
+  @Test
+  public void shouldDenyAccess() {
+    // Then:
+    expectedException.expect(AssertionError.class);
+    expectedException.expectMessage(
+        String.format("Access denied to User:%s", VALID_USER1.username)
+    );
+
+    // When:
+    makeKsqlRequest(VALID_USER1, "SHOW TOPICS;");
```

Comment:
nit: can we rename `VALID_USER1` to simply `USER1`? When reading this test I was expecting the request to succeed at first. Plus there are no invalid users in this test.

Reply:
Changed it.